server/utils/routeBuilder.ts

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable max-classes-per-file */
```

Comment:
a little fluent iterface / DSL to allow defining routes and ensuring role protection is explicitly stated and async middleware applied correctly